app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -1359,7 +1359,14 @@ private void naturalLanguageRecognition(String allWords) {
         allWords = allWords.trim();
         allWords = allWords.replaceAll(":", "."); // fix real times
         allWords = allWords.replaceAll("(\\d)([a-zA-Z])", "$1 $2"); // fix like 22mm
-        allWords = allWords.replaceAll("([0-9]\\.[0-9])([0-9][0-9])", "$1 $2"); // fix multi number order like blood 3.622 grams
```

Comment:
Commenting this line out removes the implemented fix. It looks like sometimes people speak two voice treatment commands right after another, maybe in different word/number ordering like "blood 3.6" followed by "22 grams". The voice recognition engine concatenates the recognized text to a single string without a separator between the two commands.
The regex tries to fix this but is incomplete. Here is an improved regex: `([a-zA-Z]+ \d{1,3}(?:\.\d)*)(\d+(?:\.\d)* [a-zA-Z]+)`. This handles the following cases correctly:
```
blood 3.622 grams
grams 223.6 blood
grams 1802.9 blood
grams 180114 blood
1.298 carbs
```

They will be changed to:
```
blood 3.6 22 grams
grams 22 3.6 blood
grams 180 2.9 blood
grams 180 114 blood
1.298 carbs
```

Now, there is a separator between the numbers but only if the input string starts with a word and ends with a word. The latest example is unaffected.

All examples will be processed correctly with this regex.